job/jobcache.go

```diff
@@ -1,36 +1,86 @@
 package job
 
 import (
+	"os"
+	"os/signal"
 	"sync"
+	"time"
 )
 
-var (
-	AllJobs = &JobCache{
-		Jobs:   make(map[string]*Job),
-		rwLock: sync.Mutex{},
+type JobCache interface {
+	Get(id string) *Job
+	GetAll() map[string]*Job
+	Set(j *Job)
+	Delete(id string)
+	Persist() error
+}
+
+type MemoryJobCache struct {
+	Jobs            map[string]*Job
+	rwLock          sync.Mutex
+	jobDB           JobDB
+	persistWaitTime time.Duration
+}
+
+func NewMemoryJobCache(jobDB JobDB, persistWaitTime time.Duration) *MemoryJobCache {
+	if persistWaitTime == time.Duration(0) {
+		persistWaitTime = 5 * time.Second
 	}
-)
+	return &MemoryJobCache{
+		Jobs:            map[string]*Job{},
+		rwLock:          sync.Mutex{},
+		jobDB:           jobDB,
+		persistWaitTime: persistWaitTime,
+	}
+}
+
+func (c *MemoryJobCache) Init() {
+	// Prep cache
+	allJobs, err := c.jobDB.GetAll()
+	if err != nil {
+		log.Fatal(err)
+	}
+	for _, v := range allJobs {
+		v.StartWaiting(c)
+		c.Set(v)
+	}
+
+	// Occasionally, save items in cache to db.
+	go c.PersistEvery(c.persistWaitTime)
```

Comment:
@ajvb ? How about having `c.PersistEvery()` taking no parameters, and internally using `c.persistWaitTime`?